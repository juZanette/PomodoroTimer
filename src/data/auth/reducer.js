const initialState = {
  signed: false,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SIGN_IN':
      return {
        signed: true,
      }
    default:
      return state
  }
}
