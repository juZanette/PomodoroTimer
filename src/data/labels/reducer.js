const initialState = {
  dialogOpened: false,
  labelToEdit: null,
  formValue: {
    name: null,
    color: null,
  },
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_DIALOG_OPENED':
      return {
        ...state,
        dialogOpened: action.dialogOpened,
      }
    case 'SET_LABEL_TO_EDIT':
      return {
        ...state,
        labelToEdit: action.labelToEdit,
      }
    case 'SET_FORM_VALUE':
      return {
        ...state,
        formValue: action.formValue,
      }
    default:
      return state
  }
}
