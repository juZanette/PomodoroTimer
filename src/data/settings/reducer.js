const defaultState = {
  workDuration: 25,
  shortBreakDuration: 5,
  longBreakDuration: 20,
  rounds: 4,
  showTimerInTitle: true,
  showNotifications: true,
}

const initialState = {
  workDuration: null,
  shortBreakDuration: null,
  longBreakDuration: null,
  rounds: null,
  showTimerInTitle: null,
  showNotifications: null,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_WORK_DURATION':
      return {
        ...state,
        workDuration: action.duration,
      }
    case 'SET_SHORT_BREAK_DURATION':
      return {
        ...state,
        shortBreakDuration: action.duration,
      }
    case 'SET_LONG_BREAK_DURATION':
      return {
        ...state,
        longBreakDuration: action.duration,
      }
    case 'SET_ROUNDS':
      return {
        ...state,
        rounds: action.rounds,
      }
    case 'SET_SHOW_TIMER_IN_TITLE':
      return {
        ...state,
        showTimerInTitle: action.showTimerInTitle,
      }
    case 'SET_SHOW_NOTIFICATIONS':
      return {
        ...state,
        showNotifications: action.showNotifications,
      }
    case 'SET_SETTINGS':
      return {
        ...defaultState,
        ...action.settings,
      }

    default:
      return state
  }
}
