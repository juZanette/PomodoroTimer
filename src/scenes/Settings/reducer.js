const initialState = {
  rounds: 4,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_ROUNDS':
      return {
        rounds: action.roundsNumber,
      }
    default:
      return state
  }
}
