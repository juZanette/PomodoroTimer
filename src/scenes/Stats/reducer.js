const initialState = {
  totalSessions: 0,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'INCREMENT_TOTAL_SESSIONS':
      return {
        totalSessions: ++state.totalSessions,
      }
    default:
      return state
  }
}
