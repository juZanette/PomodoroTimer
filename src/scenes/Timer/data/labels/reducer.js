const initialState = {
  dialogOpened: false,
  formDialogOpened: false,
  deleteAlert: {
    opened: false,
    labelToDelete: null,
  },
  labelToEdit: null,
  formValue: null,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_DIALOG_OPENED':
      return {
        ...state,
        dialogOpened: action.dialogOpened,
      }
    case 'SET_FORM_DIALOG':
      return {
        ...state,
        formDialogOpened: action.formDialogOpened,
      }
    case 'SET_DELETE_ALERT':
      const { opened, labelToDelete } = action.deleteAlert

      return {
        ...state,
        deleteAlert: {
          opened,
          labelToDelete,
        },
      }
    case 'SET_LABEL_TO_EDIT':
      return {
        ...state,
        labelToEdit: action.labelToEdit,
      }
    case 'SET_FORM_VALUE':
      return {
        ...state,
        formValue: action.formValue,
      }
    default:
      return state
  }
}
