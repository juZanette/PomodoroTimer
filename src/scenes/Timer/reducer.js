const initialState = {
  timerActive: false,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_TIMER_ACTIVE':
      return {
        timerActive: action.timerActive,
      }
    default:
      return state
  }
}
