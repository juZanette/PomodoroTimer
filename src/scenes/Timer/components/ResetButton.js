import React from 'react'
import IconButton from '@material-ui/core/IconButton'
import ReplayIcon from '@material-ui/icons/Replay'
import { useDispatch, useSelector } from 'react-redux'
import { resetTimer } from '../actions'
import { TYPES } from '../reducer'

const ResetButton = () => {
  const dispatch = useDispatch()
  const { type, timeLeft } = useSelector((state) => state.timer)
  const {
    workDuration,
    shortBreakDuration,
    longBreakDuration,
    showTimerInTitle,
  } = useSelector((state) => state.settings)

  const handleClick = () => {
    let duration

    switch (type) {
      case TYPES.work:
        duration = workDuration
        break
      case TYPES.shortBreak:
        duration = shortBreakDuration
        break
      case TYPES.longBreak:
        duration = longBreakDuration
        break
      default:
        break
    }
    dispatch(resetTimer({ duration, showTimerInTitle }))
  }

  return (
    <IconButton
      disabled={!timeLeft}
      aria-label="Reset timer"
      onClick={handleClick}
    >
      <ReplayIcon />
    </IconButton>
  )
}

export { ResetButton }
