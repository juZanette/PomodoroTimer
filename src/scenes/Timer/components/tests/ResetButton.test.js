import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { createMount } from '@material-ui/core/test-utils'
import IconButton from '@material-ui/core/IconButton'
import { ResetButton } from '../ResetButton'
import { TYPES, STATUSES } from '../../data/timer/reducer'
import { SaveSessionAlert } from '../SaveSessionAlert'
import { Dialog } from '@material-ui/core'

describe('<ResetButton />', () => {
  const mockStore = configureMockStore([thunk])

  let store
  let mount
  let wrapper

  beforeEach(() => {
    mount = createMount()

    wrapper = (
      status = STATUSES.onHold,
      type = TYPES.workDuration,
      timeLeft = { minutes: 25, seconds: 0 }
    ) => {
      const storeData = {
        timer: { status, type, timeLeft, saveSessionAlert: false },
        settings: {
          workDuration: 25,
          shortBreakDuration: 5,
          longBreakDuration: 20,
          showTimerInTitle: true,
        },
        labels: {
          labelSelected: null,
        },
      }

      store = mockStore(storeData)
      store.dispatch = jest.fn()

      return mount(
        <Provider store={store}>
          <ResetButton />
        </Provider>
      )
    }
  })

  test('should render <ResetButton /> correctly', () => {
    expect(wrapper()).toMatchSnapshot()
  })

  test('should be disabled if the timer is on hold', () => {
    const wrapperRendered = wrapper()
    expect(wrapperRendered.find(IconButton).prop('disabled')).toBe(true)
  })

  test('should be enabled if the timer is running', () => {
    const wrapperRendered = wrapper(STATUSES.running, TYPES.work, {
      minutes: 22,
      seconds: 10,
    })

    expect(wrapperRendered.find(IconButton).prop('disabled')).toBe(false)
  })

  test('should be enabled if the timer is paused', () => {
    const wrapperRendered = wrapper(STATUSES.paused, TYPES.work, {
      minutes: 22,
      seconds: 10,
    })

    expect(wrapperRendered.find(IconButton).prop('disabled')).toBe(false)
  })

  test('should display save session alert with the remaining time if the work timer has been running for 1 or more minutes', () => {
    const wrapperRendered = wrapper(STATUSES.running, TYPES.work, {
      minutes: 22,
      seconds: 10,
    })

    wrapperRendered.find(IconButton).simulate('click')

    expect(wrapperRendered.find(SaveSessionAlert).prop('time')).toEqual({
      minutes: 2,
      seconds: 50,
    })
  })

  test('should not display save session alert if the timer has been running for less than a minute', () => {
    const wrapperRendered = wrapper(STATUSES.running, TYPES.work, {
      minutes: 24,
      seconds: 10,
    })

    wrapperRendered.find(IconButton).simulate('click')

    expect(wrapperRendered.find(SaveSessionAlert).prop('time')).toEqual({
      minutes: 0,
      seconds: 0,
    })
  })

  test('should not display save session alert if the timer type is short break', () => {
    const wrapperRendered = wrapper(STATUSES.running, TYPES.shortBreak, {
      minutes: 3,
      seconds: 10,
    })

    wrapperRendered.find(IconButton).simulate('click')

    expect(wrapperRendered.find(SaveSessionAlert).prop('time')).toEqual({
      minutes: 0,
      seconds: 0,
    })
  })

  test('should not display save session alert if the timer type is long break', () => {
    const wrapperRendered = wrapper(STATUSES.running, TYPES.longBreak, {
      minutes: 11,
      seconds: 10,
    })

    wrapperRendered.find(IconButton).simulate('click')

    expect(wrapperRendered.find(SaveSessionAlert).prop('time')).toEqual({
      minutes: 0,
      seconds: 0,
    })
  })
})
