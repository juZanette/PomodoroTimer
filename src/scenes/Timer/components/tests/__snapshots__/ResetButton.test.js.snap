// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetButton /> should render <ResetButton /> correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ResetButton>
    <WithStyles(ForwardRef(IconButton))
      aria-label="Reset timer"
      disabled={true}
      onClick={[Function]}
    >
      <ForwardRef(IconButton)
        aria-label="Reset timer"
        classes={
          Object {
            "colorInherit": "MuiIconButton-colorInherit",
            "colorPrimary": "MuiIconButton-colorPrimary",
            "colorSecondary": "MuiIconButton-colorSecondary",
            "disabled": "Mui-disabled",
            "edgeEnd": "MuiIconButton-edgeEnd",
            "edgeStart": "MuiIconButton-edgeStart",
            "label": "MuiIconButton-label",
            "root": "MuiIconButton-root",
            "sizeSmall": "MuiIconButton-sizeSmall",
          }
        }
        disabled={true}
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(ButtonBase))
          aria-label="Reset timer"
          centerRipple={true}
          className="MuiIconButton-root Mui-disabled"
          disabled={true}
          focusRipple={true}
          onClick={[Function]}
        >
          <ForwardRef(ButtonBase)
            aria-label="Reset timer"
            centerRipple={true}
            className="MuiIconButton-root Mui-disabled"
            classes={
              Object {
                "disabled": "Mui-disabled",
                "focusVisible": "Mui-focusVisible",
                "root": "MuiButtonBase-root",
              }
            }
            disabled={true}
            focusRipple={true}
            onClick={[Function]}
          >
            <button
              aria-label="Reset timer"
              className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <ForwardRef>
                  <WithStyles(ForwardRef(SvgIcon))>
                    <ForwardRef(SvgIcon)
                      classes={
                        Object {
                          "colorAction": "MuiSvgIcon-colorAction",
                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                          "colorError": "MuiSvgIcon-colorError",
                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                          "root": "MuiSvgIcon-root",
                        }
                      }
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8z"
                        />
                      </svg>
                    </ForwardRef(SvgIcon)>
                  </WithStyles(ForwardRef(SvgIcon))>
                </ForwardRef>
              </span>
            </button>
          </ForwardRef(ButtonBase)>
        </WithStyles(ForwardRef(ButtonBase))>
      </ForwardRef(IconButton)>
    </WithStyles(ForwardRef(IconButton))>
    <SaveSessionAlert
      time={
        Object {
          "minutes": 0,
          "seconds": 0,
        }
      }
    >
      <WithStyles(ForwardRef(Dialog))
        aria-describedby="save-session-alert-description"
        aria-labelledby="save-session-alert-title"
        onClose={[Function]}
        open={false}
      >
        <ForwardRef(Dialog)
          aria-describedby="save-session-alert-description"
          aria-labelledby="save-session-alert-title"
          classes={
            Object {
              "container": "MuiDialog-container",
              "paper": "MuiDialog-paper",
              "paperFullScreen": "MuiDialog-paperFullScreen",
              "paperFullWidth": "MuiDialog-paperFullWidth",
              "paperScrollBody": "MuiDialog-paperScrollBody",
              "paperScrollPaper": "MuiDialog-paperScrollPaper",
              "paperWidthFalse": "MuiDialog-paperWidthFalse",
              "paperWidthLg": "MuiDialog-paperWidthLg",
              "paperWidthMd": "MuiDialog-paperWidthMd",
              "paperWidthSm": "MuiDialog-paperWidthSm",
              "paperWidthXl": "MuiDialog-paperWidthXl",
              "paperWidthXs": "MuiDialog-paperWidthXs",
              "root": "MuiDialog-root",
              "scrollBody": "MuiDialog-scrollBody",
              "scrollPaper": "MuiDialog-scrollPaper",
            }
          }
          onClose={[Function]}
          open={false}
        >
          <ForwardRef(Modal)
            BackdropComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "children": [Function],
                    "className": [Function],
                    "classes": [Function],
                    "invisible": [Function],
                    "open": [Function],
                    "transitionDuration": [Function],
                  },
                  "render": [Function],
                },
                "displayName": "WithStyles(ForwardRef(Backdrop))",
                "options": Object {
                  "defaultTheme": Object {
                    "breakpoints": Object {
                      "between": [Function],
                      "down": [Function],
                      "keys": Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ],
                      "only": [Function],
                      "up": [Function],
                      "values": Object {
                        "lg": 1280,
                        "md": 960,
                        "sm": 600,
                        "xl": 1920,
                        "xs": 0,
                      },
                      "width": [Function],
                    },
                    "direction": "ltr",
                    "mixins": Object {
                      "gutters": [Function],
                      "toolbar": Object {
                        "@media (min-width:0px) and (orientation: landscape)": Object {
                          "minHeight": 48,
                        },
                        "@media (min-width:600px)": Object {
                          "minHeight": 64,
                        },
                        "minHeight": 56,
                      },
                    },
                    "overrides": Object {},
                    "palette": Object {
                      "action": Object {
                        "activatedOpacity": 0.12,
                        "active": "rgba(0, 0, 0, 0.54)",
                        "disabled": "rgba(0, 0, 0, 0.26)",
                        "disabledBackground": "rgba(0, 0, 0, 0.12)",
                        "disabledOpacity": 0.38,
                        "focus": "rgba(0, 0, 0, 0.12)",
                        "focusOpacity": 0.12,
                        "hover": "rgba(0, 0, 0, 0.04)",
                        "hoverOpacity": 0.04,
                        "selected": "rgba(0, 0, 0, 0.08)",
                        "selectedOpacity": 0.08,
                      },
                      "augmentColor": [Function],
                      "background": Object {
                        "default": "#fafafa",
                        "paper": "#fff",
                      },
                      "common": Object {
                        "black": "#000",
                        "white": "#fff",
                      },
                      "contrastThreshold": 3,
                      "divider": "rgba(0, 0, 0, 0.12)",
                      "error": Object {
                        "contrastText": "#fff",
                        "dark": "#d32f2f",
                        "light": "#e57373",
                        "main": "#f44336",
                      },
                      "getContrastText": [Function],
                      "grey": Object {
                        "100": "#f5f5f5",
                        "200": "#eeeeee",
                        "300": "#e0e0e0",
                        "400": "#bdbdbd",
                        "50": "#fafafa",
                        "500": "#9e9e9e",
                        "600": "#757575",
                        "700": "#616161",
                        "800": "#424242",
                        "900": "#212121",
                        "A100": "#d5d5d5",
                        "A200": "#aaaaaa",
                        "A400": "#303030",
                        "A700": "#616161",
                      },
                      "info": Object {
                        "contrastText": "#fff",
                        "dark": "#1976d2",
                        "light": "#64b5f6",
                        "main": "#2196f3",
                      },
                      "primary": Object {
                        "contrastText": "#fff",
                        "dark": "#303f9f",
                        "light": "#7986cb",
                        "main": "#3f51b5",
                      },
                      "secondary": Object {
                        "contrastText": "#fff",
                        "dark": "#c51162",
                        "light": "#ff4081",
                        "main": "#f50057",
                      },
                      "success": Object {
                        "contrastText": "rgba(0, 0, 0, 0.87)",
                        "dark": "#388e3c",
                        "light": "#81c784",
                        "main": "#4caf50",
                      },
                      "text": Object {
                        "disabled": "rgba(0, 0, 0, 0.38)",
                        "hint": "rgba(0, 0, 0, 0.38)",
                        "primary": "rgba(0, 0, 0, 0.87)",
                        "secondary": "rgba(0, 0, 0, 0.54)",
                      },
                      "tonalOffset": 0.2,
                      "type": "light",
                      "warning": Object {
                        "contrastText": "rgba(0, 0, 0, 0.87)",
                        "dark": "#f57c00",
                        "light": "#ffb74d",
                        "main": "#ff9800",
                      },
                    },
                    "props": Object {},
                    "shadows": Array [
                      "none",
                      "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
                      "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
                      "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
                      "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
                      "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
                      "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
                      "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
                      "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
                      "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
                      "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
                      "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
                      "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
                      "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
                      "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
                      "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
                      "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
                      "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
                      "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
                      "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
                      "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
                      "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
                      "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
                      "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
                      "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
                    ],
                    "shape": Object {
                      "borderRadius": 4,
                    },
                    "spacing": [Function],
                    "transitions": Object {
                      "create": [Function],
                      "duration": Object {
                        "complex": 375,
                        "enteringScreen": 225,
                        "leavingScreen": 195,
                        "short": 250,
                        "shorter": 200,
                        "shortest": 150,
                        "standard": 300,
                      },
                      "easing": Object {
                        "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
                        "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
                        "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
                        "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
                      },
                      "getAutoHeightDuration": [Function],
                    },
                    "typography": Object {
                      "body1": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                        "fontSize": "1rem",
                        "fontWeight": 400,
                        "letterSpacing": "0.00938em",
                        "lineHeight": 1.5,
                      },
                      "body2": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                        "fontSize": "0.875rem",
                        "fontWeight": 400,
                        "letterSpacing": "0.01071em",
                        "lineHeight": 1.43,
                      },
                      "button": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                        "fontSize": "0.875rem",
                        "fontWeight": 500,
                        "letterSpacing": "0.02857em",
                        "lineHeight": 1.75,
                        "textTransform": "uppercase",
                      },
                      "caption": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                        "fontSize": "0.75rem",
                        "fontWeight": 400,
                        "letterSpacing": "0.03333em",
                        "lineHeight": 1.66,
                      },
                      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                      "fontSize": 14,
                      "fontWeightBold": 700,
                      "fontWeightLight": 300,
                      "fontWeightMedium": 500,
                      "fontWeightRegular": 400,
                      "h1": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                        "fontSize": "6rem",
                        "fontWeight": 300,
                        "letterSpacing": "-0.01562em",
                        "lineHeight": 1.167,
                      },
                      "h2": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                        "fontSize": "3.75rem",
                        "fontWeight": 300,
                        "letterSpacing": "-0.00833em",
                        "lineHeight": 1.2,
                      },
                      "h3": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                        "fontSize": "3rem",
                        "fontWeight": 400,
                        "letterSpacing": "0em",
                        "lineHeight": 1.167,
                      },
                      "h4": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                        "fontSize": "2.125rem",
                        "fontWeight": 400,
                        "letterSpacing": "0.00735em",
                        "lineHeight": 1.235,
                      },
                      "h5": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                        "fontSize": "1.5rem",
                        "fontWeight": 400,
                        "letterSpacing": "0em",
                        "lineHeight": 1.334,
                      },
                      "h6": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                        "fontSize": "1.25rem",
                        "fontWeight": 500,
                        "letterSpacing": "0.0075em",
                        "lineHeight": 1.6,
                      },
                      "htmlFontSize": 16,
                      "overline": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                        "fontSize": "0.75rem",
                        "fontWeight": 400,
                        "letterSpacing": "0.08333em",
                        "lineHeight": 2.66,
                        "textTransform": "uppercase",
                      },
                      "pxToRem": [Function],
                      "round": [Function],
                      "subtitle1": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                        "fontSize": "1rem",
                        "fontWeight": 400,
                        "letterSpacing": "0.00938em",
                        "lineHeight": 1.75,
                      },
                      "subtitle2": Object {
                        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                        "fontSize": "0.875rem",
                        "fontWeight": 500,
                        "letterSpacing": "0.00714em",
                        "lineHeight": 1.57,
                      },
                    },
                    "zIndex": Object {
                      "appBar": 1100,
                      "drawer": 1200,
                      "mobileStepper": 1000,
                      "modal": 1300,
                      "snackbar": 1400,
                      "speedDial": 1050,
                      "tooltip": 1500,
                    },
                  },
                  "name": "MuiBackdrop",
                },
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                "useStyles": [Function],
              }
            }
            BackdropProps={
              Object {
                "transitionDuration": Object {
                  "enter": 225,
                  "exit": 195,
                },
              }
            }
            className="MuiDialog-root"
            closeAfterTransition={true}
            disableBackdropClick={false}
            disableEscapeKeyDown={false}
            onClose={[Function]}
            open={false}
          />
        </ForwardRef(Dialog)>
      </WithStyles(ForwardRef(Dialog))>
    </SaveSessionAlert>
  </ResetButton>
</Provider>
`;
